packages/react-devtools-shared/src/utils.js

```diff
@@ -46,17 +46,15 @@ export function getDisplayName(
     return nameFromCache;
   }
 
-  let displayName;
+  let displayName = fallbackName;
 
   // The displayName property is not guaranteed to be a string.
   // It's only safe to use for our purposes if it's a string.
   // github.com/facebook/react-devtools/issues/803
   if (typeof type.displayName === 'string') {
     displayName = type.displayName;
-  }
-
-  if (!displayName) {
-    displayName = type.name || fallbackName;
+  } else if (typeof type.name === 'string') {
```

Comment:
The previous truthy check also prevented returning empty strings (for anonymous functions). This test would fail after this change:
```js
expect(getDisplayName(() => {})).toEqual('Anonymous');
```
I think the check should be:
```
typeof type.name === 'string' && type.name !== ''
```

Reply:
yes, you right :( It would be enough to add one more test...

Reply:
yes, you right :( It would be enough to add one more test...

Reply:
No worries, I added it~